crates/compiler/builtins/roc/Dict.roc

```diff
@@ -1446,3 +1489,29 @@ expect
     |> Dict.insert "Charlie" 19
     |> Dict.walkUntil Bool.false (\_, _, age -> if age >= 18 then Break Bool.true else Continue Bool.false)
     |> Bool.isEq Bool.true
+
```

Comment:
Just to be more thorough with the testing and index changes, can we make a test case where two neighboring keys in the middle of a dictionary are removed.

Hmm, actually since remove reorders, probably should also test, key in middle and key at end being removed.

But yeah, can we get just a few more cases?

Reply:
I thought that `remove` reorders, but I haven't ran into any bugs. I added those cases

Reply:
Since all of the tests are using `==` they unordered nature of remove is hidden. `==` does an unordered comparison of two dictionaries. If you were to walk the dictionaries or convert them to lists, the ordering changes from removal would be visible.